pkg/image/importer/importer.go

```diff
@@ -372,6 +382,75 @@ func (isi *ImageStreamImporter) calculateImageSize(ctx gocontext.Context, repo d
 	return nil
 }
 
+func manifestFromManifestList(ctx gocontext.Context, manifestList *manifestlist.DeserializedManifestList, ref imageapi.DockerImageReference, s distribution.ManifestService, defaultArch, defaultOS string) (distribution.Manifest, error) {
+	if len(manifestList.Manifests) == 0 {
+		return nil, fmt.Errorf("no manifests in manifest list %s", ref.Exact())
+	}
+
+	if defaultArch == "" {
+		defaultArch = runtime.GOARCH
+	}
+	if defaultOS == "" {
+		defaultOS = runtime.GOOS
+	}
+
+	var manifestDigest digest.Digest
+	for _, manifestDescriptor := range manifestList.Manifests {
```

Comment:
If there is one manifest, we should always select, regardless of OS and Arch.  Otherwise we are breaking backwards compatibility for clients that switch to manifest list but still only publish a single image.

Reply:
i believe that's what line 404 is handling.  (the case that we didn't manage to match anything).

Reply:
Yeah, thanks.